docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -119,29 +116,33 @@ None currently known to be needed
 
-An operator will be able to decide to turn the feature on or off with a flag on creation of a cluster, and then can alter the YAML file for AzureMachineTemplate and AzureMachinePool to customize how long they want the caching interval to be.
+Example of the enviornment variable being turned on:
 
-Every time that amount of time has passed, it should go through the nodes of the cluster, choose a healthy node, shut it down, take a snapshot of it, restart it, create a shared image gallery image, delete the snapshot, and then configure the AzureMachineTemplate specs to use that shared image gallery image.
+```
+export AZURE_OS_CACHING=true
+```
+
+The controller will maintain a timestamp, and when the current time is the chosen interval ahead or more, the controller will perform the caching. When the process is started it should go through the nodes of the cluster, choose a healthy node, shut it down, take a snapshot of it, restart it, create a shared image gallery image, delete the snapshot, and then configure the AzureMachineTemplate specs to use that shared image gallery image. After, it will store the current time as its timestamp.
 
-	As for why the healthy node has to be shut down while creating a snapshot of it, if it isn’t shut down first then pods can be scheduled as the snapshot is taken which will cause some dangerous states in terms of how it exists after being utilized by the AzureMachineTemplates.
+As for why the healthy node has to be shut down while creating a snapshot of it, if it isn’t shut down first then pods can be scheduled as the snapshot is taken which will cause some dangerous states in terms of how it exists after being utilized by the AzureMachineTemplates.
 
-In terms of how a healthy node would be selected, there should be an annotation on each node as to when they’ve last been patched. An ideal node would be one which has been patched since the last prototype went into service and is running and healthy. Whichever node has been steadily healthy for the longest amount of time should be chosen as it’s the most overall stable. As the prototype is always from a successfully healthy and working node the image is always known to be working before being chosen for replication.
+In terms of how a healthy node would be selected, there is already an annotation on each node from its status as to when they’ve last been patched. An ideal node would be one which has been patched since the last prototype went into service and is running and healthy. Whichever node has been steadily healthy for the longest amount of time since the last patch and has the patch applied should be chosen as it’s the most overall stable. As the prototype is always from a successfully healthy and working node the image is always known to be working before being chosen for replication.
```

Comment:
> there is already an annotation on each node from its status as to when they’ve last been patched

When I read "annotation" in this context that generally refers to the object's `metadata.annotations` but "status" usually means the `status` field on the object. Do you mean one of those in particular? And could you provide an example here?

Reply:
I specified and gave example YAMLs (sadly this isn't a default on AzureMachine instances so it might be out of scope for this proposal, I proposed adding these timestamps to it or we can store it programmatically and it is already present on AzureMachinePool instances)